items.py

```diff
@@ -254,14 +254,19 @@ def use_dropkick(targeted_enemy, using_card, entity):
 def use_dualwield(using_card, entity):
     '''Create a(2) copy(copies) of an Attack or Power card.'''
     while True:
-        view.view_piles(entity.hand, entity, False, 'card.get("Type") in ("Attack", "Power")')
-        option = view.list_input(f"Choose a card to make {'a copy' if not using_card.get('Upgraded') else '2 copies'} of > ", entity.hand)
+        valid_cards = [(idx, card) for idx, card in enumerate(entity.hand) if card.get('Type') in ('Attack', 'Power')]
+        view_cards = [card[1] for card in valid_cards]
+        view.view_piles(pile=view_cards, entity=entity)
+        option = view.list_input(f"Choose a card to make {'a copy' if not using_card.get('Upgraded') else '2 copies'} of > ", valid_cards)
+        #convert option to index of entity.hand
+        option = valid_cards[option][0]
```

Comment:
I don't get how the conversion works here. Can you explain this?

Reply:
```python
# enumerate returns an index and item for items in a list, so you don't have to manage a counter yourself.
# So this line will create a new list that has the indicies of the hand and the card if it's valid.
# Ex: If there's 3 cards in the hand, at indicies 0, 1, 2 -- and the middle one is invalid -- this would create the following list:
# [(0, first_card), (2, third_card)]
# Note that the indexes of of this new list are different from the original. 0 and 1 map to 0 and 2.
valid_cards = [(idx, card) for idx, card in enumerate(entity.hand) if card.get('Type') in ('Attack', 'Power')]

# We pull out the cards part of the tuple and put that up for display
view_cards = [card[1] for card in valid_cards]
view.view_piles(pile=view_cards, entity=entity)

# Then, once we select a card, we know how to map the indicies. The chosen index maps to `valid_cards `, and from there, the first part of the tuple matches the index in the hand.
option = view.list_input(..., valid_cards)